### DISCUSSION TITLE
Spectral library generation and a discrepancy in transition names

### DISCUSSION THREAD

**Discussion by sfilip-fn** [vdemichev/DiaNN#372] [Q&A]
Hi Vadim,

I have a question about the spectral library generation and transition names.

I tried to generate a library using DiaNN's _in-silico_ prediction tool and MaxQuant's msms.txt file transformed into a library using DiaNN (MaxQuant version: 2.0.3.0). I noticed a discrepancy in transition names for y-ions for the same transition, depending on the library generation method. For example:

**DiaNN's _in-silico_ prediction tool**
**transition_name column: ** AAAATGTIFTFR2_121_1_0_7
**FragmentSeriesNumber column: ** 5
**ProductMz column: ** 841.4559
Mass in ProductMz column is correct for y-ion 7, which is also correctly marked in transition_name column. This means that the FragmentSeriesNumber column is incorrect and it should be 7. Along the same lines, the FragmentSeriesNumber is reversed (12 amino acids - 5 in FragmentSeriesNumber = 7, which is the correct number).

**MaxQuant's msms.txt transformed into a library using DIANN**
**transition_name column: ** AAAATGTIFTFR2_121_1_0_7
**FragmentSeriesNumber column: ** 5
**ProductMz column: ** 683.3886
Mass in ProductMz column is correct for y ion 5, which is also correctly marked in FragmentSeriesNumber column. This means that the y-ion number in the transition_name column is incorrect/reversed and it should be 5.

This issue is only present for y-ions, while b-ions are correct and consistent.

Another issue that I noticed for MaxQuant's msms.txt file transformed into a library is that sometimes it incorrectly assigns FragmentCharge 1 to ions that are supposed to be FragmentCharge 2, as the Mz in ProductMz column is halved.

My questions are:
1. Which column is used to for the analysis in DiaNN? Transition_name and/or ProductMz and/or FragmentSeriesNumber?
2. Will the y-ion discrepancy affect the final results?
3. Will the FragmentCharge discrepancy affect the final results?
4. If I wanted to combine msms.txt derived and _in-silico_ predicted transitions, which column should I use to do it and do I need to correct the y-ions?

Thank you!
Szymon

**Comment by vdemichev** [comment]
Hi Szymon,

transition_name is never used by DIA-NN and is just a unique identifier which is not supposed to be interpreted. Strange that FragmentSeriesNumber is incorrect in the first case. Can you please describe in a bit more detail what you did to obtain this result? Is the DIA-NN log available?

MaxQuant msms.txt support is 'experimental', has not been tested properly.

Best,
Vadim

**Reply by sfilip-fn** [reply]
Hi Vadim,

Thank you for the reply.

I did the following:
1. Added a fasta file to DIANN.
2. Clicked both options for Precursor Ion generation: "Fasta digest for library free search" and "Deep learning-based spectra".
3. Added Oxidation of M as a variable modification with Maximum number of variable modifications set to 2.
4. I left all the other options as default and clicked run.
5. The produced library output was only in .speclib format, so I resubmitted the .speclib file, used the same settings as described above and clicked run. 
6. This produced a library in a .tsv format.

I attached both log files (first one is in_silico_library.log and second on e is to_tsv_from_speclib.log).

[in_silico_library.log.txt](https://github.com/vdemichev/DiaNN/files/8632423/in_silico_library.log.txt)
[to_tsv_from_speclib.log.txt](https://github.com/vdemichev/DiaNN/files/8632424/to_tsv_from_speclib.log.txt)


Thank you,
Szymon

**Reply by vdemichev** [reply]
OK, many thanks! I can't reproduce this with a different FASTA, can you please also share the test_fasta.fasta file?

Best,
Vadim

**Reply by sfilip-fn** [reply]
I was unable to provide the previous test fasta, but I got the same issue with human swissprot fasta. I used version 1.8.0 for it, but the issue exists in both versions 1.8.0 and 1.8.1.

I am sending the new log files:
- human_sp_library.log.txt to generate speclib 
- human_sp_library_to_tsv.predicted.log.txt to move speclib to tsv
- human fasta -- I had to change the format to .txt, as Github doesn't support .fasta uploads

Please let me know if anything else is missing.
Thank you!

Best,
Szymon

[human_sp_library_to_tsv.predicted.log.txt](https://github.com/vdemichev/DiaNN/files/8633808/human_sp_library_to_tsv.predicted.log.txt)
[human_sp_library.log.txt](https://github.com/vdemichev/DiaNN/files/8633809/human_sp_library.log.txt)
[Human.txt](https://github.com/vdemichev/DiaNN/files/8633810/Human.txt)

**Reply by vdemichev** [reply]
Can you please check which specific peptide shows this with this human FASTA?

**Reply by sfilip-fn** [reply]
I saw it for this peptde: AVVEVDESGTRAAAATGTIFTFR.
However, the issue is consistent across all y ion peptides.
